ledger/participant-integration-api/src/test/lib/scala/platform/store/dao/JdbcLedgerDaoPostCommitValidationSpec.scala

```diff
@@ -125,7 +125,8 @@ private[dao] trait JdbcLedgerDaoPostCommitValidationSpec extends LoneElement {
     }
   }
 
-  it should "be able to use divulged contract in later transaction" in {
+  // TODO it seems as this test would be based on flawed assumptions (a pure divulgance without create contract having a ledger_effective_time)
```

Comment:
Can this be a situation that arises in a distributed setting, with a contract created remotely being divulged to a local party?

Reply:
AFAIK there are two distinct cases for divulgence:
- public: all divulged contracts have created event in the indexer already
- private: the above is not true, but in this case the divulged contract itself is exposed (so create argument, and template id we can get this way)

Furthermore: in the private mode we do not have ledger_effective_time information of the creation of the contract. And therefore also postcommit validation is not supported.

Also: AFAIK post commit validation is only running in case daml-on-sql/sandbox, which are working in public mode.

Also: question is how can we do post-commit validation if we do not have the create event for the contract? AFAIK: the create event's ledger_effective_Time should be considered foll post commit validation

the above are working theories, let's discuss this further and align them!

Reply:
After an offline discussion with @nmarton-da, we realized this test covers an impossible case.

Post-commit validation is only performed in the "write directly to the index without ~collecting $20~ going through the ledger" (a.k.a. sandbox-classic). But in the sandbox-classic case, divulgences will _always_ be empty ([code](https://github.com/digital-asset/daml/blob/7ae28beff98df7bfd9d5368ea22d46717c0cc353/ledger/sandbox-classic/src/main/scala/platform/sandbox/stores/ledger/sql/SqlLedger.scala#L382)).

As a matter of fact, the `divulged` parameter of the post-commit validation function can be dropped entirely.